Source/SpatialGDK/Private/SpatialInterop.cpp

```diff
@@ -860,16 +867,148 @@ bool USpatialInterop::IsSingletonClass(UClass* Class)
 	return false;
 }
 
-NameToEntityIdMap* USpatialInterop::GetSingletonNameToEntityId() const
+improbable::unreal::GlobalStateManagerData* USpatialInterop::GetGlobalStateManagerData() const
 {
 	TSharedPtr<worker::View> View = SpatialOSInstance->GetView().Pin();
 	auto It = View->Entities.find(SpatialConstants::GLOBAL_STATE_MANAGER);
 
 	if (It != View->Entities.end())
 	{
 		improbable::unreal::GlobalStateManagerData* GSM = It->second.Get<improbable::unreal::GlobalStateManager>().data();
-		return &(GSM->singleton_name_to_entity_id());
+		return GSM;
 	}
 
 	return nullptr;
 }
+
+NameToEntityIdMap* USpatialInterop::GetSingletonNameToEntityId() const
+{
+	improbable::unreal::GlobalStateManagerData* GSM = GetGlobalStateManagerData();
+
+	if (GSM == nullptr)
+	{
+		return nullptr;
+	}
+
+	return &(GSM->singleton_name_to_entity_id());
+}
+
+EntityIdToPathMap* USpatialInterop::GetEntityIdToReplicatedStablyNamedPath() const
+{
+	improbable::unreal::GlobalStateManagerData* GSM = GetGlobalStateManagerData();
+
+	if (GSM == nullptr)
+	{
+		return nullptr;
+	}
+
+	return &(GSM->entity_id_to_replicated_stably_named_path());
+}
+
+void USpatialInterop::ReserveReplicatedStablyNamedActorChannel(USpatialActorChannel* Channel)
+{
+	ReplicatedStablyNamedActorChannelQueue.Add(Channel);
+
+	if (CanSpawnReplicatedStablyNamedActors())
+	{
+		ReserveReplicatedStablyNamedActors();
+	}
+}
+
+void USpatialInterop::UnreserveReplicatedStablyNamedActor(AActor* Actor)
+{
+	// Slightly inefficient linear search, could use a hashmap
+	USpatialActorChannel* Channel = nullptr;
+	for (auto& It : ReplicatedStablyNamedActorChannelQueue)
```

Comment:
TArray.find() then .removeAt(index)?

Reply:
Actually given that you only use Queue for add, remove, iteration, find, seems like TSet is ideal (if no duplicate keys)

Reply:
Yes, this is not ideal (I put inefficient next to things I don't feel comfortable writing), but in this case I need the Channel->Actor, so find will not work unless I can get the Actor's channel from somewhere, and I am not sure where.

Reply:
I rewrote the whole logic, using a TMap now. Although still not ideal, it is faster and has less code.